From d517938989ffb7e85434a2da442bef61d89bf706 Mon Sep 17 00:00:00 2001
From: Orlando Chamberlain <redecorating@protonmail.com>
Date: Tue, 11 Jan 2022 16:20:33 +1100
Subject: [PATCH 1/1] x86: efi: force runtime v1.10 fix T2 nvram writes

EFI v1.10 doesn't have EFI_QUERY_VARIABLE_INFO, so linux won't use it.
This means T2 Macs will be able to write to nvram.

Link: https://lore.kernel.org/all/20220111051717.25b86946@localhost/T/#mde6cc47d38ff5c9ad561eaf932bbd20e6e8eeca8

---
 arch/x86/platform/efi/efi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/platform/efi/efi.c b/arch/x86/platform/efi/efi.c
index 147c30a81f15..d7203355cc69 100644
--- a/arch/x86/platform/efi/efi.c
+++ b/arch/x86/platform/efi/efi.c
@@ -399,7 +399,7 @@ static int __init efi_systab_init(unsigned long phys)
 		efi_nr_tables		= systab32->nr_tables;
 	}
 
-	efi.runtime_version = hdr->revision;
+	efi.runtime_version = EFI_1_10_SYSTEM_TABLE_REVISION;
 
 	efi_systab_report_header(hdr, efi_fw_vendor);
 	early_memunmap(p, size);
-- 
2.34.1

